### PR TITLE
Creates updateSession and createSession in PaymentProviderService

### LINKED ISSUE #69: [PaymentProviderService]: Add updateSession and createSession
```
updateSession(paymentSession, cart)
createSession(providerId, cart)
```

The `paymentSession` argument in `updateSession` contains `{ provider_id, data }` allowing the payment provider service to retrieve the payment provider via `provider_id` and give the method the data needed to update by passing `data`.

The `createSession` method will call the `createSession` method of the provider identified by `providerId`.

### PR DESCRIPTION
**Pull request by srindom**
Makes it easy for the cart service to create/update sessions without having to retrieve the provider each time. Closes #69.

### CODE REVIEW

**Comment by olivermrbl** [comment]
Can I get you to pull master and push, such that I don't need to review all the files from last PR again?

**Comment by srindom** [comment]
@olivermrbl done

File: packages/medusa/src/services/payment-provider.js [hunk 1]
```diff
@@ -10,21 +10,41 @@ class PaymentProviderService {
...
-   *    data - data to send to the subscriber
-   *
+   * Creates a payment session with the given provider.
+   * @param {string} providerId - the id of the provider to create payment with
+   * @param {Cart} cart - a cart object used to calculate the amount, etc. from
+   * @return {Promise} the payment session
+   */
+  createSession(providerId, cart) {
+    const provider = this.retrieveProvider(providerId)
+    return provider.createSession(cart)
```
Comment (olivermrbl) [inline packages/medusa/src/services/payment-provider.js:20]: Should this not be `createPayment`, if we look at the current state of `BasePaymentService`?

File: packages/medusa/src/services/payment-provider.js [hunk 2]
```diff
@@ -10,21 +10,41 @@ class PaymentProviderService {
...
+  /**
+   * Updates an existing payment session.
+   * @param {PaymentSession} paymentSession - the payment session object to
+   *    update
+   * @param {Cart} cart - the cart object to update for
+   * @return {Promise} the updated payment session
+   */
+  updateSession(paymentSession, cart) {
+    const provider = this.retrieveProvider(paymentSession.provider_id)
+    return provider.updateSession(paymentSession.data, cart)
```
Comment (olivermrbl) [inline packages/medusa/src/services/payment-provider.js:32]: And `updatePayment` here

File: packages/medusa/src/services/payment-provider.js (hunk 1 above)
Comment (srindom) [reply packages/medusa/src/services/payment-provider.js:20]: You're right 🤦

**Pull request merged**